configuration/postgres/postgres.go

```diff
@@ -0,0 +1,417 @@
+	}
+	items := make(map[string]*configuration.Item)
+	for i := 0; rows.Next(); i++ {
+		var item configuration.Item
+		var key string
+		var metadata []byte
+		v := make(map[string]string)
+		if err := rows.Scan(&key, &item.Value, &item.Version, &metadata); err != nil {
+			return nil, fmt.Errorf("error in reading data from configuration store: '%s'", err)
+		}
+		if err := json.Unmarshal(metadata, &v); err != nil {
+			return nil, fmt.Errorf("error in unmarshalling response from configuration store: '%s'", err)
+		}
+		item.Metadata = v
+		if item.Value != "" {
+			items[key] = &item
+		}
+	}
+	return &configuration.GetResponse{
+		Items: items,
+	}, nil
+}
+
+func (p *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler) (string, error) {
+	var pgNotifyChannels []string
+	for k, v := range req.Metadata {
+		if res := strings.EqualFold(pgNotifyChannelKey, k) && !slices.Contains(pgNotifyChannels, v); res {
+			pgNotifyChannels = append(pgNotifyChannels, v) // append unique channel names only
+		}
+	}
+	if len(pgNotifyChannels) == 0 {
+		return "", fmt.Errorf("unable to subscribe to '%s'.pgNotifyChannel attribute cannot be empty", p.metadata.configTable)
+	}
+	return p.subscribeToChannel(ctx, pgNotifyChannels, req, handler)
+}
+
+func (p *ConfigurationStore) Unsubscribe(ctx context.Context, req *configuration.UnsubscribeRequest) error {
+	p.configLock.Lock()
+	defer p.configLock.Unlock()
+	for k, v := range p.ActiveSubscriptions {
+		if v.uuid == req.ID {
+			if oldStopChan, ok := p.subscribeStopChanMap[req.ID]; ok {
+				delete(p.subscribeStopChanMap, req.ID)
+				close(oldStopChan.(chan struct{}))
+				pgChannel := fmt.Sprintf(unlistenTemplate, k)
+				conn, err := p.client.Acquire(ctx)
+				if err != nil {
+					p.logger.Errorf("error acquiring connection:", err)
+					return fmt.Errorf("error acquiring connection: %s ", err)
+				}
+				defer conn.Release()
+				_, err = conn.Exec(ctx, pgChannel)
+				if err != nil {
+					p.logger.Errorf("error listening to channel:", err)
+					return fmt.Errorf("error listening to channel: %s", err)
+				}
+				delete(p.ActiveSubscriptions, k)
+				return nil
+			}
+		}
+	}
+
+	return fmt.Errorf("unable to find subscription with ID : %v", req.ID)
+}
+
+func (p *ConfigurationStore) doSubscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler, command string, channel string, subscription string, stop chan struct{}) {
+	conn, err := p.client.Acquire(ctx)
+	if err != nil {
+		p.logger.Errorf("error acquiring connection:", err)
+		return
+	}
+	defer conn.Release()
+	if _, err = conn.Exec(ctx, command); err != nil {
+		p.logger.Errorf("error listening to channel:", err)
+		return
+	}
+	for {
+		notification, err := conn.Conn().WaitForNotification(ctx)
+		if err != nil {
+			if !pgconn.Timeout(err) && !errors.Is(ctx.Err(), context.Canceled) {
```

Comment:
`if !pgconn.Timeout(err) && err != context.Canceled {`

Because `err` may not be `ctx.Err()`